extensions/binding/org.eclipse.smarthome.binding.sonyaudio/src/main/java/org/eclipse/smarthome/binding/sonyaudio/handler/SonyAudioHandler.java

```diff
@@ -0,0 +1,536 @@
+                updateState(SonyAudioBindingConstants.CHANNEL_VOLUME, new DecimalType(volume / 100.0));
+                break;
+            case 1:
+                updateState(SonyAudioBindingConstants.CHANNEL_ZONE1_VOLUME, new DecimalType(volume / 100.0));
+                break;
+            case 2:
+                updateState(SonyAudioBindingConstants.CHANNEL_ZONE2_VOLUME, new DecimalType(volume / 100.0));
+                break;
+            case 3:
+                updateState(SonyAudioBindingConstants.CHANNEL_ZONE3_VOLUME, new DecimalType(volume / 100.0));
+                break;
+            case 4:
+                updateState(SonyAudioBindingConstants.CHANNEL_ZONE4_VOLUME, new DecimalType(volume / 100.0));
+                break;
+        }
+    }
+
+    @Override
+    public void updateMute(int zone, boolean mute) {
+        switch (zone) {
+            case 0:
+                updateState(SonyAudioBindingConstants.CHANNEL_MUTE, mute ? OnOffType.ON : OnOffType.OFF);
+                break;
+            case 1:
+                updateState(SonyAudioBindingConstants.CHANNEL_ZONE1_MUTE, mute ? OnOffType.ON : OnOffType.OFF);
+                break;
+            case 2:
+                updateState(SonyAudioBindingConstants.CHANNEL_ZONE2_MUTE, mute ? OnOffType.ON : OnOffType.OFF);
+                break;
+            case 3:
+                updateState(SonyAudioBindingConstants.CHANNEL_ZONE3_MUTE, mute ? OnOffType.ON : OnOffType.OFF);
+                break;
+            case 4:
+                updateState(SonyAudioBindingConstants.CHANNEL_ZONE4_MUTE, mute ? OnOffType.ON : OnOffType.OFF);
+                break;
+        }
+    }
+
+    @Override
+    public void updatePowerStatus(int zone, boolean power) {
+        switch (zone) {
+            case 0:
+                updateState(SonyAudioBindingConstants.CHANNEL_POWER, power ? OnOffType.ON : OnOffType.OFF);
+                break;
+            case 1:
+                updateState(SonyAudioBindingConstants.CHANNEL_ZONE1_POWER, power ? OnOffType.ON : OnOffType.OFF);
+                break;
+            case 2:
+                updateState(SonyAudioBindingConstants.CHANNEL_ZONE2_POWER, power ? OnOffType.ON : OnOffType.OFF);
+                break;
+            case 3:
+                updateState(SonyAudioBindingConstants.CHANNEL_ZONE3_POWER, power ? OnOffType.ON : OnOffType.OFF);
+                break;
+            case 4:
+                updateState(SonyAudioBindingConstants.CHANNEL_ZONE4_POWER, power ? OnOffType.ON : OnOffType.OFF);
+                break;
+        }
+    }
+
+    private void startAutomaticRefresh(int refresh) {
+        if (refresh <= 0) {
+            return;
+        }
+
+        refreshJob = scheduler.scheduleWithFixedDelay(() -> {
+            try {
+                List<Channel> channels = getThing().getChannels();
+                for (Channel channel : channels) {
+                    handleCommand(channel.getUID(), RefreshType.REFRESH);
```

Comment:
This looks pretty inefficient for those reasons:
1. You are doing calls for channels that aren't linked at all - that's a waste of time. Better use the `callback.isChannelLinked` method.
2. This can cause a huge number of external requests; is the connection able to easily handle this load? Afaics, some requests contain data for multiple channels, so that the same requests are done multiple times. If my assumption is true, you might want to consider using the `ExpiringCache` see [here](https://www.eclipse.org/smarthome/documentation/features/frameworkUtilities.html#caching).